agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -2069,6 +2267,186 @@ bool backhaul_manager::handle_1905_higher_layer_data_message(ieee1905_1::CmduMes
+     * Create response message
+     */
+    auto cmdu_tx_header =
+        cmdu_tx.create(mid, ieee1905_1::eMessageType::LINK_METRIC_RESPONSE_MESSAGE);
+    if (!cmdu_tx_header) {
+        LOG(ERROR) << "Failed creating LINK_METRIC_RESPONSE_MESSAGE header! mid=" << std::hex
+                   << (int)mid;
+        return false;
+    }
+
+    /**
+     * If the specified neighbor 1905.1 AL ID does not identify a neighbor of the receiving 1905.1
+     * AL, then a link metric ResultCode TLV (see Table 6-21) with a value set to “invalid
+     * neighbor” shall be included in this message.
+     */
+    bool invalid_neighbor = false;
+    if (ieee1905_1::eLinkMetricNeighborType::SPECIFIC_NEIGHBOR == neighbor_type) {
+
+        // TODO: check if specified neighbor is invalid (topology database is required for that)
+    }
+
+    if (invalid_neighbor) {
+        auto tlvLinkMetricResultCode = cmdu_tx.addClass<ieee1905_1::tlvLinkMetricResultCode>();
+        if (!tlvLinkMetricResultCode) {
+            LOG(ERROR) << "addClass ieee1905_1::tlvLinkMetricResultCode failed, mid=" << std::hex
+                       << (int)mid;
+            return false;
+        }
+
+        tlvLinkMetricResultCode->value() = ieee1905_1::tlvLinkMetricResultCode::INVALID_NEIGHBOR;
+    } else if (ieee1905_1::eLinkMetricNeighborType::SPECIFIC_NEIGHBOR == neighbor_type) {
+
+        /**
+         * Name of the interface in the receiving 1905.1 AL, which connects to an interface in the neighbor
+         * 1905.1 AL.
+         */
+        std::string local_interface_name;
+
+        /**
+         * The underlying network technology of the connecting interface
+         */
+        ieee1905_1::eMediaType media_type = ieee1905_1::eMediaType::UNKNONWN_MEDIA;
+
+        if ((network_utils::mac_to_string(neighbor_al_mac) == controller_bridge_mac) &&
+            (SBackhaulConfig::EType::Wired == m_sConfig.eType)) {
+            /**
+             * If specified neighbor is the controller and it is connected through the wired backhaul
+             * interface, then interface is the wired backhaul interface.
+             */
+            local_interface_name = m_sConfig.wire_iface;
+
+            /**
+             * Compute particular Ethernet media type
+             */
+            if (!get_media_type(local_interface_name, ieee1905_1::eMediaTypeGroup::IEEE_802_3,
+                                media_type)) {
+                LOG(ERROR) << "Unable to compute media type for interface " << local_interface_name;
+                return false;
+            }
+
+        } else {
+            //TODO this is a valid neighbor so it must be connected through a wireless interface, which is not yet supported
+        }
+
+        /**
+         * Obtain link metrics and add a Transmitter Link Metric TLV or a Receiver Link Metric TLV
+         * or both to response message.
+         */
+        if (!add_link_metrics(local_interface_name, media_type, reporter_al_mac, neighbor_al_mac,
```

Comment:
Very good point. 
I've included your suggestions to simplify code. However, one-to-many mapping between interfaces and neighbors is possible in WiFi so I've kept that possibility also for Ethernet (because it's simpler to code, doesn't hurt and who knows if someday someone connects more than one IEEE 1905 device to a switch).